packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -1,21 +1,246 @@
-
 const POLLING_INTERVAL = process.env.GATSBY_PREVIEW_POLL_INTERVAL || 3000
 
-export default function Indicator({ children }) {
+export function PreviewIndicator({ children }) {
+  return (
+    <>
+      <Style />
+      <div
+        data-testid="preview-status-indicator"
+        data-gatsby-preview-indicator="root"
+        aria-live="assertive"
+      >
+        {React.Children.map(children, (child, i) =>
+          React.cloneElement(child, { ...child.props, buttonIndex: i })
+        )}
+      </div>
+    </>
+  )
+}
+
+export function GatsbyIndicatorButton(props) {
+  return <IndicatorButton testId="gatsby" iconSvg={gatsbyIcon} {...props} />
+}
+
+const copySuccessTooltip = (
+  <>
+    {successIcon}
+    {`Link copied`}
+  </>
+)
+
+export function LinkIndicatorButton(props) {
+  const { orgId, siteId, buildId, active } = props
+  const [linkButtonCopyProps, setLinkButtonCopyProps] = useState()
+
+  const copyLinkClick = () => {
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_CLICK`,
+      orgId,
+      siteId,
+      buildId,
+      name: `copy link`,
+    })
+
+    setLinkButtonCopyProps({
+      tooltipContent: copySuccessTooltip,
+      overrideShowTooltip: true,
+    })
+
+    setTimeout(() => {
+      setLinkButtonCopyProps({
+        tooltipContent: copySuccessTooltip,
+        overrideShowTooltip: false,
+      })
+      // We want the tooltip to linger for two seconds to let the user know it has been copied
+    }, 2000)
+
+    setTimeout(() => {
+      setLinkButtonCopyProps({ tooltipContent: `Copy Link` })
+      // The tooltips fade out, in order to make sure that the text does not change
+      // while it is fading out we need to wait a bit longer than the time used above.
+    }, 2400)
+
+    if (window) {
+      navigator.clipboard.writeText(window.location.href)
+    }
+  }
+
+  const trackHover = () => {
+    if (!active) return
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_HOVER`,
+      orgId,
+      siteId,
+      buildId,
+      name: `link hover`,
+    })
+  }
+
+  return (
+    <IndicatorButton
+      testId={`link`}
+      iconSvg={linkIcon}
+      onClick={copyLinkClick}
+      onMouseEnter={trackHover}
+      {...props}
+      {...linkButtonCopyProps}
+    />
+  )
+}
+
+export function InfoIndicatorButton(props) {
+  const { active, orgId, siteId, buildId } = props
+  const trackHover = () => {
+    if (!active) return
+    trackEvent({
+      eventType: `PREVIEW_INDICATOR_HOVER`,
+      orgId,
+      siteId,
+      buildId,
+      name: `info hover`,
+    })
+  }
+  return (
+    <IndicatorButton
+      testId="info"
+      iconSvg={infoIcon}
+      onMouseEnter={trackHover}
+      {...props}
+    />
+  )
+}
+
+export function BuildErrorIndicatorTooltip({ siteId, orgId, buildId }) {
+  return (
+    <>
+      {failedIcon}
+      {`Unable to build preview`}
+      <a
+        href={generateBuildLogUrl({ orgId, siteId, buildId })}
+        target="_blank"
+        rel="noreferrer"
```

Comment:
Are we sure we don't want a referral?

Reply:
Yeah, I guess this doesn't matter since we're linking to `gatsbyjs.com`

Reply:
On second thought, I don't know if we necessarily care if we have it, so I'm going to leave unless there is good reason to remove it.